src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1265,28 +1269,89 @@ public Page search(String q) throws Exception {
     }
 
     /**
-     * Get JSON from A Jenkins endpoint.
-     * @param path The endpoint URL.
-     * @return The JSON.
+     * Get JSON from a Jenkins relative endpoint. Create a new default webclient. If you want to configure the
+     * webclient, for example to set a token for authentication, or accept other HTTP responses than 200, you can use
+     * {@link JenkinsRule#getJSON(String, WebClient)} .
+     * @param path relative path, should not start with '/'
+     * @return The JSON response from server.
      */
     public JSONWebResponse getJSON(@NonNull String path) throws IOException {
-        assert !path.startsWith("/");
 
         JenkinsRule.WebClient webClient = createWebClient();
-        Page runsPage = null;
-        try {
-            runsPage = webClient.goTo(path, "application/json");
-        } catch (SAXException e) {
-            // goTo shouldn't be throwing a SAXException for JSON.
-            throw new IllegalStateException("Unexpected SAXException.", e);
-        }
-        WebResponse webResponse = runsPage.getWebResponse();
+        return getJSON(path, webClient);
+    }
+
+    /**
+     * Get JSON from a Jenkins relative endpoint.
+     * You can preconfigure webClient for example to set a token for authentication, or accept error HTTP status.
+     * @param path relative path, should not start with '/'
+     * @param webClient a pre-configured web client.
```

Comment:
Indicates that these should be instance methods on `WebClient`.

Reply:
I don't understand what you are asking here, is it a link to java classes ?

Reply:
In general if you have a method

```java
class A {
  ReturnType someMethod(B b, String otherArg) {
    // does not use `this`
  }
}
class B {
}
```

that is a sign you should consider switching the receiver

```java
class A {
}
class B {
  ReturnType someMethod(String otherArg) {
    // uses `this`
  }
}
```

Reply:
it's not going to be a breaking change if I move them all ? Including the existing one ?

Reply:
I was referring to new methods introduced in this PR, not existing methods in `JenkinsRule` with signatures

```java
public JSONWebResponse getJSON(String path);
public JSONWebResponse postJSON(String path, Object json);
```

Reply:
> In summary, as a matter of API design I would recommend
> 
>     * moving new methods to use `WebClient` as the receiver rather than an argument
> 
>     * drop `User` parameter from `postJSON`

Ok I think I get it now, thanks for detailed explanations. I wanted to imitate the existing method but it was no the best idea. I am going to rework on that.

Reply:
done.